maas/resource_maas_zone_test.go

```diff
@@ -0,0 +1,118 @@
+package maas_test
+
+import (
+	"fmt"
+	"strings"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccResourceMaasZone_basic(t *testing.T) {
+
+	var zone entity.Zone
+	name := acctest.RandomWithPrefix("tf-zone-")
+	description := "Test description"
+
+	checks := []resource.TestCheckFunc{
+		testAccMaasZoneCheckExists("maas_zone.test", &zone),
+		resource.TestCheckResourceAttr("maas_zone.test", "name", name),
+		resource.TestCheckResourceAttr("maas_zone.test", "description", description),
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccCheckMaasZoneDestroy,
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMaasZone(name, description),
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+			// Test import using name
+			{
+				ResourceName: "maas_zone.test",
+				ImportState:  true,
+				ImportStateCheck: func(is []*terraform.InstanceState) error {
+					var zone *terraform.InstanceState
+					if len(is) != 1 {
+						return fmt.Errorf("expected 1 state: %#v", t)
+					}
+					zone = is[0]
+					assert.Equal(t, zone.Attributes["name"], name)
+					assert.Equal(t, zone.Attributes["description"], description)
+					return nil
+				},
+			},
+		},
+	})
+}
+
+func testAccMaasZoneCheckExists(rn string, zone *entity.Zone) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("resource not found: %s\n %#v", rn, s.RootModule().Resources)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("resource id not set")
+		}
+
+		conn := testutils.TestAccProvider.Meta().(*client.Client)
+		gotZone, err := conn.Zone.Get(rs.Primary.ID)
```

Comment:
This is also needed if we are to save the database ID as the Terraform resource ID

```go
                id, err := strconv.Atoi(rs.Primary.ID)
		if err != nil {
			return err
		}
```